pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -389,18 +392,39 @@ func (a *Assignment) findFlavorForResourceGroup(
 			}
 		}
 
+		if shouldTryNextFlavor(representativeMode, cq.FlavorFungibility, whetherNeedBorrowing) {
+			return assignments, nil
+		}
 		if representativeMode > bestAssignmentMode {
 			bestAssignment = assignments
 			bestAssignmentMode = representativeMode
-			if bestAssignmentMode == Fit {
```

Comment:
why is it ok to remove this?

Reply:
we will handle this case in shouldTryNextFlavor like:
```
	if representativeMode == Fit && whetherNeedBorrowing && policyBorrow == v1beta1.Borrow {
		return false
	}

	if representativeMode == Fit && !whetherNeedBorrowing {
		return false
	}
```

Reply:
When representativeMode == Fit and whetherNeedBorrowing == true and policyBorrow == v1beta1.Borrow, we will visit next flavor so bestAssignmentMode == Fit should not be break the loop